controllers/profile.js

```diff
@@ -0,0 +1,96 @@
+import db from '../models';
+
+const { users } = db;
+
+/**
+* @exports ProfileController
+* @class ProfileController
+* @description Handles User Profile
+* */
+class ProfileController {
+  /**
+  * Get all user profiles by username
+  * @async
+  * @param  {object} req - Request object
+  * @param {object} res - Response object
+  * @param {object} next The next middleware
+  * @return {json} Returns json object
+  * @static
+  */
+  static async getProfile(req, res, next) {
+    try {
+      const { username } = req.params;
+      const user = await users.findOne({
+        where: { username }
+      });
+
+      if (!user) {
+        return res.status(404).json({
+          error: 'User does not exists!',
+        });
+      }
+      const profile = await user.getProfile();
+      return res.status(200).json({
+        message: 'User profile retrieved!',
+        profile
+      });
+    } catch (error) {
+      next(error);
```

Comment:
Sometimes this error is a very verbose sequelize error, its best to capture all error codes sent on this block and send a custom error message